bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -1179,12 +1196,12 @@ public void run() {
     public int shutdown() {
         return shutdown(ExitCode.OK);
     }
-
+    private final AtomicBoolean shuttingDown = new AtomicBoolean(false);
```

Comment:
fixed

Reply:
After checking again, we can't use the `shutdownTriggered`. Since if `triggerBookieShutdown` invokes `shutdown`, `shutdownTriggered` will use twice, and will block `shutdown`